src/RealTimeClient.js

```diff
@@ -0,0 +1,174 @@
+import JWT from 'jwt-client';
+import * as messages from './subscriptions/messages';
+
+const WEBSOCKET_READY_STATES = {
+  CONNECTING: 0,
+  OPEN: 1,
+};
+
+/**
+ * A client that can be used for creating, maintaining, and monitoring WebSocket connections to the
+ * Track Real Time API.
+ */
+class RealTimeClient {
+
+  /**
+   * Creates a new realtime client
+   * @param {Client} client Instance of pre-configured REST client
+   * @param {Object} options Options for using the client to connect to the Track API
+   * @param {boolean} [options.reconnectOnClose] Whether to re-establish a connection when the
+   * current connection is closed.  Defaults to true.
+   * @param {string} [options.realTimeUri] The URI for the WebSocket connection to the RealTime API.
+   * Defaults to production.
+   */
+  constructor(client, options = {}) {
+    if (!client) {
+      throw new Error('Argument "client" is not specified');
+    }
+    this.client = client;
+    this.options = {
+      realTimeUri: 'wss://track-api.syncromatics.com/1/realtime',
+      reconnectOnClose: true,
+      ...options,
+    };
+
+    // dummy object for safe calling semantics later
+    this.connection = {};
+
+    this.queuedMessages = [];
+
+    // TODO: these will (probably) be used when implementing subscriptions
```

Comment:
This still todo?

Reply:
Thanks.  Subscriptions will be implemented in an upcoming story.  I'll change this comment (and the other TODO) to mention that.